src/module_wrap.cc

```diff
@@ -640,6 +640,10 @@ Maybe<URL> Resolve(Environment* env,
                    const std::string& specifier,
                    const URL& base,
                    PackageMainCheck check_pjson_main) {
+  if (!env->access_control()->has_permission(AccessControl::fsRead)) {
```

Comment:
wouldn't it make more sense to put this check where the actual fs calls happen? resolution might do a lot of other things not on the fs.

Reply:
Hm yeah, that’s what I did for other the modules. The difficulty here is that this code has a hard-to-follow call graph and I didn’t want to run a risk of missing something … plus, it would require passing on the `Environment*` pointer, which we currently don’t.

I’ll try it but this might just need some more refactoring.